src/Button/RedliningButton/RedliningButton.jsx

```diff
@@ -0,0 +1,672 @@
+    * @type {OlInteractionSelect}
+    * @private
+    */
+    _selectInteraction = null;
+
+  /**
+   * The properties.
+   * @type {Object}
+   */
+  static propTypes = {
+    /**
+     * The className which should be added.
+     *
+     * @type {String}
+     */
+    className: PropTypes.string,
+
+    /**
+     * Instance of OL map this component is bound to.
+     *
+     * @type {OlMap}
+     */
+    map: PropTypes.instanceOf(OlMap).isRequired,
+
+    /**
+     * Whether the line, point, polygon, circle, rectangle or text shape should
+     * be drawn.
+     *
+     * @type {String}
+     */
+    drawType: PropTypes.oneOf(['Point', 'LineString', 'Polygon', 'Circle', 'Rectangle', 'Text']),
+
+    /**
+     * Whether the redlining feature should be deleted, copied or modified.
+     * be drawn.
+     *
+     * @type {String}
+     */
+    editType: PropTypes.oneOf(['Copy', 'Edit', 'Delete']),
+
+    /**
+     * Name of system vector layer which will be used for redlining features.
+     *
+     * @type {String}
+     */
+    redliningLayerName: PropTypes.string,
+
+    /**
+     * Fill color of the redlining feature.
+     *
+     * @type {String}
+     */
+    fillColor: PropTypes.string,
+
+    /**
+     * Stroke color of the redlining feature.
+     *
+     * @type {String}
+     */
+    strokeColor: PropTypes.string,
+
+    /**
+     * Fill color of selected redlining feature.
+     *
+     * @type {String}
+     */
+    selectFillColor: PropTypes.string,
+
+    /**
+     * Stroke color of selected redlining feature.
+     *
+     * @type {String}
+     */
+    selectStrokeColor: PropTypes.string,
+
+    /**
+     * Title for modal used for input of labels for redlining features.
+     *
+     * @type {String}
+     */
+    modalPromptTitle: PropTypes.string,
+
+    /**
+     * Text string for `OK` button of the modal.
+     *
+     * @type {String}
+     */
+    modalPromptOkButtonText: PropTypes.string,
+
+    /**
+     * Text string for `Cancel` button of the modal.
+     *
+     * @type {String}
+     */
+    modalPromptCancelButtonText: PropTypes.string
+  };
+
+
+  /**
+   * The default properties.
+   * @type {Object}
+   */
+  static defaultProps = {
+    redliningLayerName: 'react-geo_redlining',
+    fillColor: 'rgba(154, 26, 56, 0.5)',
+    strokeColor: 'rgba(154, 26, 56, 0.8)',
+    selectFillColor: 'rgba(240, 240, 90, 0.5)',
+    selectStrokeColor: 'rgba(220, 120, 20, 0.8)',
+    modalPromptTitle: 'Label',
+    modalPromptOkButtonText: 'Ok',
+    modalPromptCancelButtonText: 'Cancel'
+  }
+
+  /**
+   * Creates the RedliningButton.
+   *
+   * @constructs RedliningButton
+   */
+  constructor(props) {
+
+    super(props);
+
+    if (!this.props.drawType && !this.props.editType) {
+      Logger.warn('Neither "drawType" nor "editType" was provided. Redlining ' +
+      'button won\'t work properly!');
+    }
+
+    this.state = {
+      redliningLayer: null,
+      interaction: null,
```

Comment:
Name is misleading as it contains an array of interactions.

Reply:
You're so right, renamed it in d9df6bd